pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,190 @@
-  ];
+  frontend = stdenv.mkDerivation {
+    pname = "dorion-ui";
+    inherit version src;
+
+    offlineCache = fetchYarnDeps {
+      yarnLock = "${src}/yarn.lock";
+      hash = "sha256-tEIaEQqnnrbJMMPICVAdU0EQ6AxhLJ3xhGczUQoG1W0=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      prefetch-yarn-deps
+      yarn
+    ];
+
+    env.ESBUILD_BINARY_PATH = lib.getExe (esbuild.overrideAttrs (final: _: {
+      version = "0.18.20";
+      src = fetchFromGitHub {
+        owner = "evanw";
+        repo = "esbuild";
+        rev = "v${final.version}";
+        hash = "sha256-mED3h+mY+4H465m02ewFK/BgA1i/PQ+ksUNxBlgpUoI=";
+      };
+      vendorHash = "sha256-mED3h+mY+4H465m02ewFK/BgA1i/PQ+ksUNxBlgpUoI=";
+    }));
+
+    postPatch = ''
+      substituteInPlace package.json --replace 'pnpm prebuild' 'true'
+    '';
+
+    configurePhase = ''
+      runHook preConfigure
+
+      export HOME=$(mktemp -d)
+      yarn config --offline set yarn-offline-mirror $offlineCache
+      fixup-yarn-lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+
+      runHook postConfigure
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      cp ${shelter} src-tauri/injection/browser.js
+      yarn --offline build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      cp -r src-tauri/{injection,html} $out
+
+      runHook postInstall
+    '';
+  };
+
+in
+
+rustPlatform.buildRustPackage {
+  inherit version src pname;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "rsrpc-0.5.1" = "sha256-U5+86B/X5zQnlairXZ2mEaBlYx5q/w4Ko5kebv3ejDM=";
+      "tauri-plugin-window-state-0.1.0" = "sha256-AQt2cJ3ZP0Ffme2lfThcjSUA6FDE6srJryaNeMRXpz0=";
+    };
+  };
 
   nativeBuildInputs = [
+    pkg-config
     autoPatchelfHook
-    dpkg
     wrapGAppsHook
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk
     gst_all_1.gst-plugins-bad
     gst_all_1.gst-plugins-base
     gst_all_1.gst-plugins-good
-    webkitgtk
+    glib-networking
   ];
 
-  installPhase = ''
-    runHook preInstall
+  runtimeDependencies = [
+    libappindicator
+    libayatana-appindicator
+  ];
 
-    mkdir -pv $out
-    mv -v {bin,lib,share} $out
+  env.TAURI_RESOURCE_DIR = "${placeholder "out"}/lib";
 
-    runHook postInstall
+  # This sed script changes tauri.conf.json file in three ways:
+  # 1. It changes the "beforeBuildCommand" field from "pnpm build" to an empty string.
+  #    This is useful to disable the pre-build command that runs.
+  # 2. It updates the "distDir" field from "https://discord.com/app" to "src".
+  #    This is to change the distribution directory from a URL to src.
+  # 3. It removes any lines containing the string "updater*".
+  #    This is used for patching out the updater.
```

Comment:
This is redundant and can now be dropped. This was proposed in my prior review. Reading the review feedback in full will save both you and the reviewer time in the future ;)

```suggestion
```

Reply:
Done